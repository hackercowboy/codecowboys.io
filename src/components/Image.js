import React from 'react';
import PropTypes from 'prop-types';
import { LazyLoadImage } from 'react-lazy-load-image-component';

function Image({
  src, placeholderSrc, alt, height, width, visibleByDefault,
}) {
  return (
    <div className="image">
      <LazyLoadImage
        alt={alt}
        height={height}
        src={src}
        placeholderSrc={placeholderSrc}
        effect="blur"
        width={width}
        visibleByDefault={visibleByDefault}
      />
    </div>
  );
}

Image.propTypes = {
  src: PropTypes.string.isRequired,
  placeholderSrc: PropTypes.string,
  alt: PropTypes.string.isRequired,
  height: PropTypes.number,
  width: PropTypes.number,
  visibleByDefault: PropTypes.bool,
};

Image.defaultProps = {
  placeholderSrc: undefined,
  height: undefined,
  width: undefined,
  visibleByDefault: false,
};

export default Image;
