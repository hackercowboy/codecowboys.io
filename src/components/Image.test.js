import React from 'react';
import { shallow } from 'enzyme';

import Image from './Image';

describe('<Image/>', () => {
  it('should render', () => {
    const wrapper = shallow(<Image src="http://example.com/image/test.png" alt="Test" />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
