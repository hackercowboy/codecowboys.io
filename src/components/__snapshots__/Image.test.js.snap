// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image/> should render 1`] = `
"<div className=\\"image\\">
  <t alt=\\"Test\\" height={[undefined]} src=\\"http://example.com/image/test.png\\" placeholderSrc={{...}} effect=\\"blur\\" width={[undefined]} visibleByDefault={false} afterLoad={[Function: afterLoad]} beforeLoad={[Function: beforeLoad]} delayMethod=\\"throttle\\" delayTime={300} threshold={100} useIntersectionObserver={true} wrapperClassName=\\"\\" />
</div>"
`;
